spec/index.bs

```diff
@@ -6659,19 +6705,17 @@ dictionary GPUQuerySetDescriptor : GPUObjectDescriptorBase {
 };
 </script>
 
-<dl dfn-type=dict-member dfn-for=GPUQuerySetDescriptor>
-    : <dfn>pipelineStatistics</dfn>
-    ::
-        The set of {{GPUPipelineStatisticName}} values in this sequence defines which pipeline statistics will be returned in the new query set.
+<div class=validusage dfn-for=GPUQuerySetDescriptor>
+    <dfn abstract-op>GPUQuerySetDescriptor Valid Usage</dfn>
 
-        <div class=validusage dfn-for=GPUQuerySetDescriptor.pipelineStatistics>
-            <dfn abstract-op>Valid Usage</dfn>
+    Given a {{GPUQuerySetDescriptor}} |this| the following validation rules apply:
 
-            1. |pipelineStatistics| is ignored if type is not {{GPUQueryType/pipeline-statistics}}.
-            2. If {{GPUFeatureName/pipeline-statistics-query}} is not available, |type| must not be {{GPUQueryType/pipeline-statistics}}.
-            3. If |type| is {{GPUQueryType/pipeline-statistics}},  |pipelineStatistics| must be a sequence of {{GPUPipelineStatisticName}} values which cannot be duplicated.
-        </div>
-</dl>
+    1. |this|.{{GPUQuerySetDescriptor/type}} must be a [=valid=] {{GPUQueryType}}.
+    1. |this|.{{GPUQuerySetDescriptor/type}} must not be {{GPUQueryType/pipeline-statistics}} if {{GPUFeatureName/pipeline-statistics-query}} is not available.
+    1. |this|.{{GPUQuerySetDescriptor/type}} must not be {{GPUQueryType/timestamp}} if {{GPUFeatureName/timestamp-query}} is not available.
+    1. The maximum limitation of |this|.{{GPUQuerySetDescriptor/count}} is 8192.
```

Comment:
Can we have an entry in GPULimits for this constant?

Reply:
As the discussion in https://github.com/gpuweb/gpuweb/issues/1072#issuecomment-696358676, this limitation should be large enough for most usage scenarios. If there is no demand for requesting a larger limit, I think we can have a hard limit here. WDYT?

Reply:
I think it's okay. Nit: reword to say "this.count must be `&le;` 8192".

Reply:
Ok, and we can always add it later if we need to.